plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiSmokeTest.java

```diff
@@ -698,7 +698,42 @@ public void testDynamicFilterEnabled_withPartitionPruningUsingDynamicFilter(Reso
                 .isEqualTo(expectedInputRowsAfterFiltering);
 
         // Exercise query and check output
-        assertQuery(query, "SELECT * FROM VALUES (1, 'a1', 100.0, 1000, 'SG'), (3, 'a3', 101.0, 1001, 'SG'), (1, 'a1', 100.0, 1000, 'SG'), (3, 'a3', 101.0, 1001, 'SG')");
+        assertQuery(query, "VALUES (1, 'a1', 100.0, 1000, 'SG'), (3, 'a3', 101.0, 1001, 'SG'), (1, 'a1', 100.0, 1000, 'SG'), (3, 'a3', 101.0, 1001, 'SG')");
+    }
+
+    @Test
+    public void testDynamicFilterEnabled_withPartitionPruningUsingDynamicFilterOnNestedPartitions()
+    {
+        Session session = SessionBuilder.from(getSession())
+                .withDynamicFilterTimeout("10s")
+                .build();
+        final String tableIdentifier = "hudi:tests." + HUDI_MULTI_PT_V8_MOR.getRoTableName();
+        // Query is joined-on recordKey and partitionField
+        @Language("SQL") String query = "SELECT t1.id FROM " +
+                HUDI_MULTI_PT_V8_MOR + " t1 " +
+                "INNER JOIN " + HUDI_MULTI_PT_V8_MOR + " t2 ON t1.id = t2.id AND t1.part_int = t2.part_int " +
+                "WHERE t2.part_int = 2023";
+
+        MaterializedResult explainRes = getQueryRunner().execute(session, "EXPLAIN ANALYZE " + query);
+        Pattern scanFilterInputRowsPattern = getScanFilterInputRowsPattern(tableIdentifier);
+        Matcher matcher = scanFilterInputRowsPattern.matcher(explainRes.toString());
```

Comment:
nit: we may want to rewrite the check for `scanFilterInputRowsPattern` to be based on the JSON parsing using `findNodeInPlan`, so such logic is unified.

Reply:
Got it